### PR TITLE
Alignment

### PR DESCRIPTION
**Pull request by IchordeDionysos**
I started to try #19 out initially only with support for text paragraphs and blockquotes.

I just want to get feedback from you what do you think about my implementation.
Yes, I know this could and should be a lot cleaner, just wanted to get things to work first.

Here is, what it would look like:
Screen Shot 2019-06-21 at 22 26 54

This is how you would set the alignment for the above example:
```
Markdown(
  data: _markdownData,
  styleSheet: MarkdownStyleSheet.fromTheme(Theme.of(context)).copyWith(
    textAlign: WrapAlignment.center,
    blockquoteAlign: WrapAlignment.end,
  ),
);
```

Additional alignments could be:
- `h1Align`
- `h2Align`
- `h3Align`
- `h4Align`
- `h5Align`
- `h6Align`
- `listAlign` or `unorderedListAlign`/`ordererdListAlign` (I would prefer the 2nd, for finer control over the alignment)
- `codeblockAlign`
- Also from above: `blockquoteAlgin`
- And: `textAlgin` (I am thinking about renaming it to `paragraphAlign`, to match the HTML name for this kind of block)

The default for all of these values would be `WrapAlignment.start` so no one is required to specify all of these if they don't need it.

Maybe we could abstract the WrapAlignment to `MarkdownAlignment` with the values `start`, `center` and `end` to not confuse anyone unfamiliar with how the Markdown widget positions elements and also to remove the possibility of specifying `WrapAlignment.spaceAround`/..., which I think in this case can be really confusing to use.

@DaveShuckerow regarding your comment about how the starting bullet would be aligned:
I looked at the way Google Docs would align the bullet and they center the bullet if the whole thing would be centered, so I would do the same thing here.
Also, I think with my current approach this would automatically be the case.

What do you think about the general interface and the way I applied the alignment?
Please don't judge me yet, for the concrete way of implementing my ideas 😄

### CODE REVIEW

**Comment by IchordeDionysos** [comment]
@DaveShuckerow Any updates/feedback from you?

**Comment by IchordeDionysos** [comment]
@DaveShuckerow I have implemented it now completely as suggested by me.
Works for all the cases I have tested.

**Comment by maks** [comment]
@IchordeDionysos  is the "more complete example" commit really needed, it seems to make this PR huge compared to the actual changes?

@DaveShuckerow is there any feedback on what is required to get this PR merged?

**Comment by IchordeDionysos** [comment]
@maks Ist seems like I have added the example folder I used to the the changes to the commit.

The important bits of the pull request involve being able to specify alignment for all important markdown types, like source code, blockquotes, lists, ...

**Comment by domesticmouse** [comment]
Can you please update this PR to fix the failing checks?

**Comment by maks** [comment]
@IchordeDionysos I think removing the example code would fix test failures, though looks like you will also need to rebase to fix up some merge conflicts on the latest commit in master.

**Comment by IchordeDionysos** [comment]
@domesticmouse I've resolved the merge conflicts and now the checks are all good

File: lib/src/builder.dart [hunk 1]
```diff
@@ -363,11 +373,55 @@ class MarkdownBuilder implements md.NodeVisitor {
...
+          textAlign: _textAlignForBlockTag(_currentBlockTag),
         ));
       } else {
         mergedTexts.add(child);
       }
     }
     return mergedTexts;
   }
+
+  TextAlign _textAlignForBlockTag(String blockTag) {
```
Comment (KagurazakaHanabi) [inline lib/src/builder.dart]: Why not use `TextAlign`

File: lib/src/builder.dart [hunk 2]
```diff
@@ -363,11 +373,55 @@ class MarkdownBuilder implements md.NodeVisitor {
...
+      case WrapAlignment.center: return TextAlign.center;
+      case WrapAlignment.end: return TextAlign.end;
+      case WrapAlignment.spaceAround: return TextAlign.justify;
+      case WrapAlignment.spaceBetween: return TextAlign.justify;
+      case WrapAlignment.spaceEvenly: return TextAlign.justify;
+    }
+    return TextAlign.start;
+  }
+
+  WrapAlignment _wrapAlignmentForBlockTag(String blockTag) {
```
Comment (KagurazakaHanabi) [inline lib/src/builder.dart]: Why not use `TextAlign`

**Comment by domesticmouse** [comment]
Please resolve conflicts

**Comment by maks** [comment]
@IchordeDionysos if you don't have time to keep working on this at the moment, I can have a go at rebasing the PR to fix merge conflicts and adding some tests.
I also fixed a small bug with ordered lists alignment: https://github.com/simpleclub/flutter_markdown/pull/2

**Comment by IchordeDionysos** [comment]
@maks We are very busy here right now, so I would be glad for any help :)

**Comment by maks** [comment]
:+1: @IchordeDionysos I'll start working on it and raise a PR against your repo.

**Comment by maks** [comment]
@IchordeDionysos I've got the PR ready now:
https://github.com/simpleclub/flutter_markdown/pull/3
it updates your branch to the latest master here and added a test for alignment.

**Comment by maks** [comment]
@googlebot I consent.

**Comment by maks** [comment]
@domesticmouse I'm not familiar with CirrusCi but from looking at the stacktrace on the test failure, it seems to be failing due to some internal error https://cirrus-ci.com/task/6125387818205184
Is the CI known to be a bit flaky? Would it be possible to just kick it off again?

**Comment by maks** [comment]
oh and there is a Travis build and it reports green so not sure why there are 2 CI's being used?

**Pull request merged**